Sources/MapboxDirections/Lane.swift

```diff
@@ -9,29 +9,54 @@ struct Lane: Equatable {
      */
     let indications: LaneIndication
     
+    /**
+     Whether the lane can be taken to complete the maneuver (`true`) or not (`false`)
+     */
     var isValid: Bool
     
-    init(indications: LaneIndication, valid: Bool = false) {
+    /**
+     Whether the lane is a preferred lane (`true`) or not (`false`)
+     A preferred lane is a lane that is recommended if there are multiple lanes available
+     */
+    var isActive: Bool?
+    
+    /**
+     Which of the `indications` is applicable to the current route, when there is more than one
+     */
+    var validIndication: LaneIndication?
+    
+    init(indications: LaneIndication, valid: Bool = false, active: Bool? = false, preferred: LaneIndication? = nil) {
         self.indications = indications
         self.isValid = valid
+        self.isActive = active
+        self.validIndication = preferred
     }
 }
 
 extension Lane: Codable {
     private enum CodingKeys: String, CodingKey {
         case indications
         case valid
+        case active
+        case preferred = "valid_indication"
     }
     
     func encode(to encoder: Encoder) throws {
         var container = encoder.container(keyedBy: CodingKeys.self)
         try container.encode(indications, forKey: .indications)
         try container.encode(isValid, forKey: .valid)
+        try container.encodeIfPresent(isActive, forKey: .active)
+        let str = validIndication?.descriptions.first
+        try container.encodeIfPresent(str, forKey: .preferred)
     }
     
     init(from decoder: Decoder) throws {
         let container = try decoder.container(keyedBy: CodingKeys.self)
         indications = try container.decode(LaneIndication.self, forKey: .indications)
         isValid = try container.decode(Bool.self, forKey: .valid)
+        isActive = try container.decodeIfPresent(Bool.self, forKey: .active)
+        if let str = try container.decodeIfPresent(String.self, forKey: .preferred) {
+            validIndication = LaneIndication(descriptions: [str])
+        }
```

Comment:
This is a bit of a workaround for the fact that LaneIndication always corresponds to an array of values in JSON, whereas we’re using the type here out of convenience for roundtripping one “description” element within such an array. To make the use of lane indication descriptions more apparent, we can name the variable a bit more verbosely:

```suggestion
        if let validIndicationDescriptions = try container.decodeIfPresent(String.self, forKey: .preferred) {
            validIndication = LaneIndication(descriptions: [validIndicationDescriptions])
        }
```